src/wallet/wallet.cpp

```diff
@@ -4007,21 +4007,44 @@ bool CWallet::ApplyMigrationData(MigrationData& data, bilingual_str& error)
         }
     }
 
+    // Pair external wallets with their corresponding db handler
+    std::vector<std::pair<std::shared_ptr<CWallet>, std::unique_ptr<WalletBatch>>> wallets_vec;
+    for (const auto& ext_wallet : {data.watchonly_wallet, data.solvable_wallet}) {
+        if (!ext_wallet) continue;
+
+        std::unique_ptr<WalletBatch> batch = std::make_unique<WalletBatch>(ext_wallet->GetDatabase());
+        if (!batch->TxnBegin()) {
+            error = strprintf(_("Error: database transaction cannot be executed for wallet %s"), ext_wallet->GetName());
+            return false;
+        }
+        wallets_vec.emplace_back(ext_wallet, std::move(batch));
+    }
+
+    // Write address book entry to disk
+    auto func_store_addr = [](WalletBatch& batch, const CTxDestination& dest, const CAddressBookData& entry) {
```

Comment:
In commit "wallet: addressbook migration, batch db writes" (342c45f80e32b0320829ce380b5854844cd74bc8)

`write_address_book` would be a more descriptive name than `func_store_addr` and `watchonly_wallets` would be more descriptive than `wallet_vec`. The `_vec` suffix and and `func_` prefix do not do anything to help a reader of the code, IMO.